src/routes/data-decode/__tests__/create-data-decoded.dto.factory.ts

```diff
@@ -0,0 +1,9 @@
+import { faker } from '@faker-js/faker';
+import { CreateDataDecodedDto } from '../entities/create-data-decoded.dto';
+
+export default function (data?: string, to?: string): CreateDataDecodedDto {
+  return <CreateDataDecodedDto>{
```

Comment:
True, I was probably thinking this was an interface. I will change it 👍🏻

Reply:
> Giving this a second thought... to instantiate the class directly we would need a constructor in the class. This is the first DTO in the project and maybe we need an alignment here... I left the DTO into the _routes_ layer because is never passed in to the _domain_ layer (`DataDecodedRepository` in this case), which receives primitives only.

Let's go with the constructor. I have a branch where I also have a constructor for it: https://github.com/5afe/safe-client-gateway-nest/blob/088dfca3d2e5b8ba16fe33a0a37051b13eccc060/src/routes/safes/entities/safe-info.entity.ts

> That's the reason why the class `CreateDataDecodedDto` is annotated with OpenAPI stuff without implementing any _domain_ entity. This pattern is sightly different, so I was thinking in moving DTOs to the _domain_, and I think _services_ could be moved as well (in another PR) to _domain_, what do you think @fmrsabino?

I don't think this is the way to go. We would be moving the OpenApi schema to domain and I think the domain should not be aware of it. I see `CreateDataDecodedDto` as a route-only entity – it only encapsulates the request payload.

Reply:
Ok, I'm fine with keeping the DTO into the routes layer, just thinking in having a counterpart in the domain layer as the rest of the entities (to keep OpenAPI definitions away from domain). Based on how the project evolves and where we place the business logic we could think if _services_ are part of the routes or domain layer. I mean, the project is small and maybe is a futile decision, but I would expect to have _services_ available even if we implement another entry layer aside REST (cli, websockets, etc.), in that case I would move them to _domain_.

Reply:
I created a constructor as you did, it's simpler and helps to keep in mind this is a route-only entity (at least for now) 👍 

Changed in [5bfc6006c96742d8521aa23768926e18aa9eb55f](https://github.com/5afe/safe-client-gateway-nest/pull/132/commits/5bfc6006c96742d8521aa23768926e18aa9eb55f)